src/client/types.ts

```diff
@@ -45,14 +52,59 @@ export class AptosApiError extends Error {
 
   readonly request: AptosRequest;
 
-  constructor(request: AptosRequest, response: AptosResponse<any, any>, message: string) {
-    super(message);
+  constructor({ apiType, aptosRequest, aptosResponse }: AptosApiErrorOpts) {
```

Comment:
this is not a breaking change since the constructor of AptosApiError is an SDK internal API detail.

Reply:
the class is exported, and might be used by external projects, so the constructor might also be used, and we need to have everything it uses exported and mentioned as changed

Reply:
why would external projects instantiate this class? This sounds like a non-intended usage of the SDK. I marked the constructor as @internal as per https://github.com/microsoft/TypeScript/issues/58891#issuecomment-2203609250 to make this clear.

Reply:
We had issues in the past where SDK users use exported functions, even if meant to use as internal. Since it is not an expected use case, we dont need to go the deprecated warning road but just mention it on CHANGELOG that the constructor interface has changed

Reply:
I still think it's a bit unnecessary and noisy to add changes to clearly internally meant signatures to the changelog. If someone does indeed indeed instantiate an AptosApiError outside of the SDK they are clearly doing something outside of the norm/advanced use-case and in those rare cases I would expect that they have a sufficient sophisticated setup and understanding of TypeScript and TypeScript would tell them that the signature has changed.

I'm happy to add this as a one-time change to the changelog for now but I think in future we should more clearly draw a line between the external API surface and internal type signatures as stabilizing and adding every single internal function signature change to a changelog sounds extremely burdensome as an SDK developer and leads to unnecessary noisy changelogs for the SDK consumer.